### PR TITLE
Fix for issue #913

### PR DESCRIPTION
**Pull request by eachimei**

### CODE REVIEW

File: client/testing/test_upload.py [hunk 1]
```diff
@@ -111,6 +111,9 @@ def test_vcs_export(self, uploadhub, repo, setupdir, tmpdir, monkeypatch):
             assert newrepo.join(repo.basename).join(".hg").listdir()
         else:
             assert newrepo.join(repo.basename).join(".git").listdir()
+        with uploadhub.workdir() as uploadbase:
+            exported = checkout.export(uploadbase)
```
Comment (fschulze) [inline client/testing/test_upload.py:115]: You should make a file read-only here. The test should then fail if you don't add the ``onerror``.

**Review by fschulze — CHANGES_REQUESTED** [review]
Could you add a news snippet in ``client/news``, see ``examples.rst`` there.

File: client/testing/test_upload.py (hunk 1 above)
Comment (eachimei) [reply client/testing/test_upload.py:116]: Done

**Comment by eachimei** [comment]
> Could you add a news snippet in `client/news`, see `examples.rst` there.

Done

**Comment by eachimei** [comment]
@fschulze I have problems running the tests locally, so I'm looking forward to see the unit-tests results as part of the CI

**Comment by fschulze** [comment]
It seems on macOS files are deleted regardless of the read only state. I'm merging as is.

**Comment by eachimei** [comment]
Thank you @fschulze 😀

**Pull request merged**